File: core/src/com/mygdx/game/handler/StopWayHandler.java
package com.mygdx.game.handler;


import com.mygdx.game.actions.BuildAction;
import com.mygdx.game.actions.BuyLandAction;
import com.mygdx.game.actions.GodMagicAction;
import com.mygdx.game.actions.LandSurpriseAction;
import com.mygdx.game.actions.PayLandAction;
import com.mygdx.game.entity.LandPoint;
import com.mygdx.game.entity.WayPoint;

/**
 * 停下来之后
 * 检测路点有没有有意外收获
 * 检测神灵是否需要加成
 * 检测是否付钱，加盖，购买，进入npc路点
 */
public class StopWayHandler extends BaseHandler {

    @Override
    public BaseHandler.HandlerEntity doHandle(BaseHandler.HandlerEntity s, HandlerChain chain) {
        s = waitAction(new LandSurpriseAction(), s);
        LandPoint landPoint = s.getPlayer().getCurrent().getLandPoint();
        System.out.println(s.getPlayer().getCurrent().getLandPoint().toString());
        if (!landPoint.isNothing()) {
            if (landPoint.isBlank()) {
                //询问购买
                waitAction(new BuyLandAction(), s);
            } else if (s.getPlayer().getName().equals(landPoint.getOwnerName())) {
                if (landPoint.isCanLevelUp()) {
                    waitAction(new BuildAction(), s);
                }
            } else if (!s.getPlayer().getName().equals(landPoint.getOwnerName())) {
                waitAction(new PayLandAction(), s);
            }
        }

        waitAction(new GodMagicAction(), s);
        return chain.process(s);
    }

}

File: core/src/main/kotlin/com/rich/diy/game/handler/StopWayHandler.kt
package com.rich.diy.game.handler

import com.rich.diy.game.entity.LandPoint
import com.rich.diy.game.entity.WayPoint
import com.rich.diy.game.events.BaseEvent

class StopWayHandler : BaseHandler() {
    override fun doHandle(s: HandlerEntity, chain: HandlerChain): HandlerEntity? {
        val current: WayPoint = s.player!!.getCurrent()
// 路点的事物
//然后处理接下来要对土地的操作
        val landPoint: LandPoint = current.landPoint
        println("停留处的关联land：" + landPoint)
        if (landPoint.isNothing) {
            //该路点对应不是土地模块
            s.parcelData = ParcelData.Builder()
                .setPrevHandlerName(this.javaClass.simpleName)
                .setTargetHandlerName(FinishRoundHandler::class.java.simpleName)
                .build()
        } else {
            if (landPoint.isBlank) {
                s.parcelData = ParcelData.Builder()
                    .setPrevHandlerName(this.javaClass.simpleName)
                    .setTargetHandlerName(BuyLandHandler::class.java.simpleName)
                    .build()
            } else {
                if (s.player!!.getName().equals(landPoint.ownerName)) {
                    s.parcelData = ParcelData.Builder()
                        .setPrevHandlerName(this.javaClass.simpleName)
                        .setTargetHandlerName(BuildLandHandler::class.java.simpleName)
                        .build()
                } else {
                    s.parcelData = ParcelData.Builder()
                        .setPrevHandlerName(this.javaClass.simpleName)
                        .setTargetHandlerName(PayLandHandler::class.java.simpleName)
                        .build()
                }
            }
        }
        return chain.process(s)
    }


}